### DISCUSSION TITLE
[Fixed] Forge test error [FAIL. Reason: Invalid data] setUp() (gas: 0)

### DISCUSSION THREAD

**Discussion by johnpaulcas** [foundry-rs/foundry#3677] [Help] [answered]
I created new project today (11/13/2022) using forge install, and try to run

```shell
> forge clean
> forge compile
```

when I tried to run 
```
> forge test
```

I encounter an error saying
```Failed tests:
[FAIL. Reason: Invalid data] setUp() (gas: 0)```

I created before some project using forge but never encounter this issue, anyone can help me to resolve this issue?


Thanks!

**Comment by johnpaulcas** [comment, marked as answer]
Found the solution by doing the following

Update faundry by running
```shell
> foundryup
```

Then initialize the project
```shell
> forge init
```

After that you need to go to faundry.toml in your root project directory then specify the `solc` verson that you are using

then run 
```shell
> forge clean
> forge build
> forge test
```

**Comment by Dliteofficial** [comment]
Followed the instructions above but I am still experiencing the same issue

**Comment by dbeal-eth** [comment]
we found that running `foundryup` is important for the instructions above.

therefore if you are getting this error, your version of foundry is most likely too old.
